### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by aditya-selldo**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ruby "3.1.0"
+
+# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem "rails", "~> 7.0.4"
+
+# The original asset pipeline for Rails [https://github.com/rails/sprockets-rails]
+gem "sprockets-rails"
+
+# Use postgresql as the database for Active Record
+gem "pg", "~> 1.1"
```
Comment (sarang-selldo) [inline Gemfile:13]: I think we will not require pg. It is configured in engine.

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "from@example.com"
+  layout "mailer"
```
Comment (sarang-selldo) [inline app/mailers/application_mailer.rb:3]: Will not require this file. You can remove it

File: app/models/application_record.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
+  primary_abstract_class
```
Comment (sarang-selldo) [inline app/models/application_record.rb:2]: We will not require this file you can remove this. Later on we inherit engines application record.

File: app/views/layouts/mailer.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
+    </style>
+  </head>
+
+  <body>
```
Comment (sarang-selldo) [inline app/views/layouts/mailer.html.erb:10]: Please remove this file

File: app/views/layouts/mailer.text.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<%= yield %>
```
Comment (sarang-selldo) [inline app/views/layouts/mailer.text.erb:1]: Please remove this file

File: config/application.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+require "active_job/railtie"
+require "active_record/railtie"
+# require "active_storage/engine"
+require "action_controller/railtie"
+require "action_mailer/railtie"
+# require "action_mailbox/engine"
+# require "action_text/engine"
+require "action_view/railtie"
+# require "action_cable/engine"
+require "rails/test_unit/railtie"
```
Comment (sarang-selldo) [inline config/application.rb:15]: Please remove this line.

File: config/initializers/content_security_policy.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# Be sure to restart your server when you modify this file.
+
+# Define an application-wide content security policy.
+# See the Securing Rails Applications Guide for more information:
+# https://guides.rubyonrails.org/security.html#content-security-policy-header
+
+# Rails.application.configure do
+#   config.content_security_policy do |policy|
```
Comment (sarang-selldo) [inline config/initializers/content_security_policy.rb:8]: Please remove this file as we are not using it.

File: config/initializers/permissions_policy.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Define an application-wide HTTP permissions policy. For further
+# information see https://developers.google.com/web/updates/2018/06/feature-policy
+#
```
Comment (sarang-selldo) [inline config/initializers/permissions_policy.rb:3]: Same here

File: test/application_system_test_case.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
+
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
```
Comment (sarang-selldo) [inline test/application_system_test_case.rb:3]: Please remove the whole test folder as we are going to configure rspec later. Please use skip option while creating project from next time

**Review by sarang-selldo — CHANGES_REQUESTED** [review]

File: app/mailers/application_mailer.rb (hunk 2 above)
Comment (aditya-selldo) [reply app/mailers/application_mailer.rb]: Done

File: app/models/application_record.rb (hunk 3 above)
Comment (aditya-selldo) [reply app/models/application_record.rb]: Done

File: app/views/layouts/mailer.html.erb (hunk 4 above)
Comment (aditya-selldo) [reply app/views/layouts/mailer.html.erb]: Done

File: app/views/layouts/mailer.text.erb (hunk 5 above)
Comment (aditya-selldo) [reply app/views/layouts/mailer.text.erb]: Done

File: config/application.rb (hunk 6 above)
Comment (aditya-selldo) [reply config/application.rb]: Done

File: config/initializers/content_security_policy.rb (hunk 7 above)
Comment (aditya-selldo) [reply config/initializers/content_security_policy.rb]: Done

File: config/initializers/permissions_policy.rb (hunk 8 above)
Comment (aditya-selldo) [reply config/initializers/permissions_policy.rb]: Done

File: test/application_system_test_case.rb (hunk 9 above)
Comment (aditya-selldo) [reply test/application_system_test_case.rb]: Done

**Review by sarang-selldo — APPROVED** [review]

**Review by sarang-selldo — APPROVED** [review]

**Review by sarang-selldo — APPROVED** [review]

**Comment by jaydip-makwana-kylas** [comment]
Add .ruby-gemset file.

**Review by jaydip-makwana-kylas — APPROVED** [review]

**Review by jaydip-makwana-kylas — APPROVED** [review]

**Pull request merged**